Mathlib/MeasureTheory/Measure/MeasureSpace.lean

```diff
@@ -2971,6 +2971,18 @@ theorem Measure.le_of_add_le_add_left [IsFiniteMeasure μ] (A2 : μ + ν₁ ≤
   fun S B1 => ENNReal.le_of_add_le_add_left (MeasureTheory.measure_ne_top μ S) (A2 S B1)
 #align measure_theory.measure.le_of_add_le_add_left MeasureTheory.Measure.le_of_add_le_add_left
 
+lemma Measure.add_left_cancel (μ ν₁ ν₂ : Measure α) [IsFiniteMeasure μ] :
+    μ + ν₁ = μ + ν₂ ↔ ν₁ = ν₂ := by
+  refine ⟨fun h ↦ ?_, fun h ↦ by rw [h]⟩
+  ext s -
+  rw [← ENNReal.add_right_inj (measure_ne_top μ s), ← Pi.add_apply, ← OuterMeasure.coe_add,
+    ← add_toOuterMeasure, h, add_toOuterMeasure, OuterMeasure.coe_add, Pi.add_apply]
```

Comment:
Can you make this an `IsRightCancelAdd` instance instead? (and would you mind reviewing #7582 which adds more of the same instances?)

Reply:
nevermind, that obviously doesn't work because `IsFiniteMeasure` is on one of the assumptions